### PR TITLE
Implement client methods for Transfer API

### PR DESCRIPTION
**Pull request by duduribeiro**
This commit introduces two client methods for the new features in
TransferAPI:
- getDomainTransfer - Retrieves the information for a specific domain
transfer
- cancelDomainTransfer - Cancels an in progress domain transfer

It also adds the new status_description field in the DomainTransfer
response.

### CODE REVIEW

File: src/main/java/com/dnsimple/data/DomainTransfer.java [hunk 1]
```diff
@@ -51,6 +54,8 @@ public boolean hasWhoisPrivacy() {
     return whoisPrivacy;
   }
 
+  public String getStatusDescription() { return statusDescription; }
```
Comment (ggalmazor) [inline src/main/java/com/dnsimple/data/DomainTransfer.java:57]:
```suggestion
  public String getStatusDescription() { 
    return statusDescription; 
  }
```

Comment (ggalmazor) [reply src/main/java/com/dnsimple/data/DomainTransfer.java:57]: I'm suggesting this to have it conforming to the rest of methods in the class. I'm guessing IntelliJ did that when automatic reformating the file. You can disable that by unchecking the `Simple blocks in one line` code style setting:
image

File: src/test/resources/com/dnsimple/cancelDomainTransfer/success.http [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+X-RateLimit-Limit: 2400
+X-RateLimit-Remaining: 2397
+X-RateLimit-Reset: 1588193270
+ETag: W/"8404bf2739e44e9f69f3a5199466776d"
+Cache-Control: no-store, must-revalidate, private, max-age=0
+X-Request-Id: 10d75bcf-0066-4d38-98a2-dc18dd5b6f4c
+X-Runtime: 1.752154
+Strict-Transport-Security: max-age=31536000
+
+{
```
Comment (ggalmazor) [inline src/test/resources/com/dnsimple/cancelDomainTransfer/success.http:17]: This is weird. Does our API pretty format JSON documents? I've checked that this is what we have at https://github.com/dnsimple/dnsimple-developer/blob/master/fixtures/v2/api/cancelDomainTransfer/success.http but it feels weird anyway.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+### master
```
Comment (ggalmazor) [inline CHANGELOG.md:3]: Thanks for tracking this here. I think I should add other stuff that I've changed too...

File: src/test/resources/com/dnsimple/cancelDomainTransfer/success.http (hunk 2 above)
Comment (weppos) [reply src/test/resources/com/dnsimple/cancelDomainTransfer/success.http:17]: goot point @ggalmazor . How did you generate those files @duduribeiro ? Did you edit them manually?

Comment (duduribeiro) [reply src/test/resources/com/dnsimple/cancelDomainTransfer/success.http:17]: @ggalmazor @weppos it is how my rest client plugin returned the data. It returned formatted for me.

Comment (weppos) [reply src/test/resources/com/dnsimple/cancelDomainTransfer/success.http:17]: We use `curl -i -s ...`. I suppose we'll have to document it anyways, actually create a shell script to make sure we all use the same approach, as per conversation with @ggalmazor related to the current issue of transfer chunked.

**Pull request merged**